src/eventHandlers/SplitsSetEvent/processLinkedIdentitySplits.ts

```diff
@@ -39,6 +46,46 @@ export async function setLinkedIdentityFlag(
     linkedIdentity.ownerAccountId,
   );
 
+  assertIsRepoDriverId(accountId);
+
+  const existingSplits = await SplitsReceiverModel.findAll({
+    where: { senderAccountId: accountId },
+    transaction,
+  });
+
+  // ORCID accounts should have exactly one splits receiver (100% to owner).
+  if (existingSplits.length > 1) {
+    const errorMsg = `Found ${existingSplits.length} splits receivers for ORCID account ${accountId}, expected 1`;
+    scopedLogger.bufferMessage(errorMsg);
+    throw new Error(errorMsg);
+  }
+
+  if (existingSplits.length > 0) {
+    await SplitsReceiverModel.destroy({
+      where: { senderAccountId: accountId },
+      transaction,
+    });
+  }
+
+  // Create the ORCID splits record (100% to owner).
+  await createSplitReceiver({
+    scopedLogger,
+    transaction,
+    splitReceiverShape: {
+      senderAccountType: 'linked_identity',
+      senderAccountId: accountId,
+      receiverAccountType: 'address',
+      receiverAccountId: linkedIdentity.ownerAccountId,
+      relationshipType: 'identity_owner',
+      weight: 1_000_000, // 100% in Drips weight format.
+      blockTimestamp,
+    },
+  });
```

Comment:
doesn't this run even if `isLinked` is false, and thus the split might not actually be 100% or exist at all?

Reply:
Good catch! Making changes.